railties/test/generators/app_generator_test.rb

```diff
@@ -677,7 +677,7 @@ def test_web_console
 
   def test_generation_runs_bundle_install
     generator([destination_root])
-    run_generator_instance
+    run_generator_and_bundler [destination_root]
```

Comment:
I'm unclear on the reason for this change.  This test is simply asserting that `bundle install` is invoked.  There doesn't seem to be a need to actually run Bundler.  As I mentioned in https://github.com/rails/rails/pull/47437#issuecomment-1489269862, it would be better to avoid running Bundler when possible for test performance reasons.

Although I should point out that `run_generator_and_bundler` isn't actually running Bundler here.  The call to `generator([destination_root])` on line 679 [sets the `@generator` variable using the given args](https://github.com/rails/rails/blob/f11c6ac45c0bc2a9a3af333a0c6cca97bac62675/railties/lib/rails/generators/testing/behavior.rb#L82).  So the call to `generator` with `"--no-skip-bundle"` inside `run_generator_and_bundler` is ignored.  In other words, __if__ this change is fixing something, it is due to stubbing `rails_gemfile_entry`.  If that is the case, we should extract a helper for that and use that instead.

Reply:
Thanks for your review!

I've made [fa86b88](https://github.com/rails/rails/pull/49679/commits/fa86b88d44248bdab1e977d4419dee064a801699) as another approach which stubs things and only tests commands we expect to be called and that's it.